README.md

```diff
@@ -83,7 +83,7 @@ Currently the easiest way to install Home Manager is as follows:
 4.  Create the first Home Manager generation:
 
     ```console
-    $ nix-shell $HM_PATH -A install --run 'home-manager switch'
+    $ $(nix-build $HM_PATH --no-out-link)/bin/home-manager switch
```

Comment:
I did a quick survey in the Nix room (#nix:matrix.org) and it was pretty scattered which of these that people preferred. Since there wasn't any clear winner I'd suggest to keep it they way it is to avoid any issues with backwards compatibility.

Reply:
Actually, part of the reason of this PR ist that the `install` attribute [introduced](https://github.com/rycee/home-manager/commit/bf3a8c63838c998e06e6825ac80f1a9a7f735a57) a month ago breaks backwards compatibility. 
For the longest time, the canonical way of package loading (`homeManager = callPackage /my/home-manager {};`) was supported.
Because the the nix-shell/build incantation mostly affects new deployments (existing users were advised to use overlays or `programs.home-manager`), the change shouldn't be disruptive.

Reply:
Yeah, I broke backwards compatibility then and that was pretty darn unfortunate. I added a news entry for the change but did so at the same time as the change was made, which was way too abrupt. Had I done it today I would have added the news entry and then waited a month before actually making the change. Fortunately, I think it did not cause any major problems for anybody (at least nobody raised an issue at the time).

In any case, for simplifying the install command @danbst made an interesting suggestion in the Nix channel. His idea was to use `shellHook` to run the install command. With something like
```nix
{
  # …

  install =
    pkgs.runCommand
      "home-manager-install"
      {
        propagatedBuildInputs = [ home-manager ];

        shellHook = ''
          echo "Creating initial Home Manager generation..."
          exec home-manager switch
        '';
      }
      "";

  # …
}
```

one would simply use `nix-shell $HM_PATH -A install` to install which I think is pretty nice! It is also compatible with the current way of installing. Does this seem OK to you?

Reply:
Thanks for taking the effort to explain it all! ☺️ 
My main reason for this quick PR was just to restore the canonical way to access the main package derivation. For obscure reasons, this helps me personally in my setup; plus, it's least suprising for the user and it simplifies the code.

We could also do this in backwards-compatible way, while sacrificing the simplicity bonus:
```nix
{ pkgs ? import <nixpkgs> {} }:
let
  home-manager = import ./home-manager {
    inherit pkgs;
    path = toString ./.;
  };
in
pkgs.stdenv.mkDerivation (home-manager.drvAttrs // {
  passthru.install = pkgs.runCommand
    "home-manager-install"
    { buildInputs = [ home-manager ]; }
    "";
})
```

Reply:
Would you mind sharing your use case where this change helps you?

Reply:
I use nix lists of directories that - when imported - each result in a derivation that can be used in 'systemPackages'.
These lists are additionally used to sync the directories to the machines that import them.
In this scheme, home-manager can't be handled like other packages because its default.nix doesn't evaluate to the main package derivation.

Sorry for bothering you with this long discussion about a rather irrelevant change.
It's okay if you close this and move on.

Reply:
Ok, and you only want to install the home-manager command? Perhaps could do
```patch
diff --git a/home-manager/default.nix b/home-manager/default.nix
index 394f0e7..90a8418 100644
--- a/home-manager/default.nix
+++ b/home-manager/default.nix
@@ -1,4 +1,4 @@
-{ pkgs
+{ pkgs ? import <nixpkgs> {}
 
   # Extra path to Home Manager. If set then this path will be tried
   # before `$HOME/.config/nixpkgs/home-manager` and
```
then I suppose it would be possible for you to continue using a directory (just would need to add an extra `/home-manager` at the end).

Otherwise my suggestion would be to use `programs.home-manager.enable = true;` in your HM configuration and let HM install itself 🙂

If you need to deploy HM configurations to other machines then I'd suggest to try the NixOS module in https://github.com/rycee/home-manager/pull/97. I'm using that with local NixOS containers and remote hosts managed by NixOps. This is not thoroughly tested, however.

Reply:
1. That would only sync `home-manager/home-manager` and leave out `home-manager/modules`.:slightly_smiling_face:
2. I have to manage a few non-NixOS machines.

I'm closing this for now. We can re-discuss when someone else complains.